src/Stack/Hoogle.hs

```diff
@@ -75,49 +80,64 @@ hoogleCmd (args,setup,rebuild,startServer) =
     hoogleMinIdent =
         PackageIdentifier hooglePackageName hoogleMinVersion
     installHoogle :: RIO EnvConfig (Path Abs File)
-    installHoogle = requiringLatestHoogle $ do
+    installHoogle = requiringHoogle Muted $ do
         Stack.Build.build Nothing
         mhooglePath' <- findExecutable "hoogle"
         case mhooglePath' of
             Right hooglePath -> parseAbsFile hooglePath
             Left _ -> do
                 logWarn "Couldn't find hoogle in path after installing.  This shouldn't happen, may be a bug."
                 bail
-    requiringLatestHoogle f = do
-        hooglePackageIdentifier <- do
-          mversion <- getLatestHackageVersion YesRequireHackageIndex hooglePackageName UsePreferredVersions
-
-          -- FIXME For a while, we've been following the logic of
-          -- taking the latest Hoogle version available. However, we
-          -- may want to instead grab the version of Hoogle present in
-          -- the snapshot current being used instead.
-          pure $ fromMaybe (Left hoogleMinIdent) $ do
-            pir@(PackageIdentifierRevision _ ver _) <- mversion
-            guard $ ver >= hoogleMinVersion
-            Just $ Right pir
-
-        case hooglePackageIdentifier of
-            Left{} -> logInfo $
-              "Minimum " <>
-              fromString (packageIdentifierString hoogleMinIdent) <>
-              " is not in your index. Installing the minimum version."
-            Right ident -> logInfo $
-              "Minimum version is " <>
-              fromString (packageIdentifierString hoogleMinIdent) <>
-              ". Found acceptable " <>
-              display ident <>
-              " in your index, installing it."
+    requiringHoogle :: Muted -> RIO EnvConfig x -> RIO EnvConfig x
+    requiringHoogle muted f = do
+        hoogleTarget <- do
+          sourceMap <- view $ sourceMapL . to smDeps
+          case Map.lookup hooglePackageName sourceMap of
+            Just hoogleDep ->
+              case dpLocation hoogleDep of
+                PLImmutable pl -> do
+                  T.pack . packageIdentifierString <$>
+                      restrictMinHoogleVersion muted (packageLocationIdent pl)
+                PLMutable _ -> do
+                  muteableLogInfo muted "Local hoogle package found, installing it"
+                  pure $ T.pack (packageNameString hooglePackageName)
+            Nothing -> do
+              mpir <- getLatestHackageVersion YesRequireHackageIndex hooglePackageName UsePreferredVersions
```

Comment:
What do you think about emitting a warning here to inform that grabbing the latest Hoogle may not work?